src/parser/jobs/mnk/modules/TwinSnakes.tsx

```diff
@@ -0,0 +1,226 @@
+		if (action.onGcd) {
+			switch (action.id) {
+			// Ignore TS itself, plus Form Shift and maybe 6SS?
+			// We use gcdsSinceTS because we don't want to double count FPF
+			case (ACTIONS.TWIN_SNAKES.id):
+				if (this.twinSnake && !this.twinSnake.end && this.gcdsSinceTS < TWIN_SNAKES_CYCLE_LENGTH) {
+					this.earlySnakes++
+				}
+
+			// Ignore Form Shift, theres probably forced downtime so we expect TS to get weird anyway
+			case (ACTIONS.FORM_SHIFT):
+				break
+
+			// Count FPF, but check if it's a bad one
+			case (ACTIONS.FOUR_POINT_FURY.id):
+				if (!this.combatants.selected.hasStatus(STATUSES.TWIN_SNAKES.id)) {
+					this.failedFury++
+				}
+
+			// Verify the window isn't closed, and count the GCDs
+			default:
+				if (this.twinSnake && !this.twinSnake.end) {
+					this.twinSnake.casts.push(event)
+				}
+
+				this.gcdsSinceTS++
+			}
+		}
+	}
+
+	// Only happens from TS itself
+	// This might be better checking if the GCD before it was buffed but ehh
+	private onGain(event: BuffEvent): void {
+		if (this.twinSnake && this.twinSnake.end) {
+			const unbuffedGcds = this.gcdsSinceTS - this.twinSnake.gcds
+			const unbuffedTime = event.timestamp - this.twinSnake.end
+			// TODO: some kind of downtime check, maybe a warning for non-GL4
+			if (unbuffedGcds > 1 || unbuffedTime > TWIN_SNAKES_CYCLE_BUFFER) {
+				this.lateSnakes++
+			}
+		}
+
+		// Start a new window
+		this.twinSnake = new TwinState(event.timestamp)
+		this.gcdsSinceTS = 0
+	}
+
+	// Can be TS or FPF - just reset the GCD count
+	// This is ok even with the reduced duration from FPF provided user is in GL3/4
+	private onRefresh(): void {
+		this.gcdsSinceTS = 0
+	}
+
+	private onDrop(event: BuffEvent): void {
+		this.stopAndSave(event.timestamp)
+	}
+
+	private onComplete() {
+		// Close off the last window
+		this.stopAndSave(this.parser.fight.end_time)
+
+		// Calculate derped potency to early refreshes
+		const lostTruePotency = this.earlySnakes * (ACTIONS.TRUE_STRIKE.potency - ACTIONS.TWIN_SNAKES.potency)
+
+		this.checklist.add(new Rule({
+			name: <Trans id="mnk.twinsnakes.checklist.name">Keep Twin Snakes up</Trans>,
+			description: <Trans id="mnk.twinsnakes.checklist.description">Twin Snakes is an easy 10% buff to your DPS across the board.</Trans>,
+			displayOrder: DISPLAY_ORDER.TWIN_SNAKES,
+			requirements: [
+				new Requirement({
+					name: <Trans id="mnk.twinsnakes.checklist.requirement.name"><ActionLink {...ACTIONS.TWIN_SNAKES} /> uptime</Trans>,
+					percent: () => this.getBuffUptimePercent(STATUSES.TWIN_SNAKES.id),
+				}),
+			],
+		}))
+
+		this.suggestions.add(new TieredSuggestion({
+			icon: ACTIONS.TWIN_SNAKES.icon,
+			content: <Trans id="mnk.twinsnakes.suggestions.early.content">
+				Avoid refreshing <ActionLink {...ACTIONS.TWIN_SNAKES} /> signficantly before its expiration as you're losing uses of the higher potency <ActionLink {...ACTIONS.TRUE_STRIKE} />.
+			</Trans>,
+			tiers: {
+				1: SEVERITY.MEDIUM,
+				4: SEVERITY.MAJOR,
+			},
```

Comment:
I'd say 1 minor, 3 medium, 5 major

Reply:
Any particular reasoning? It's a decent potency loss even just once, and 4 times is enough to noticeably effect DPS output. I'm fine with changing it, I think I chose 4 for major because that's the point where you've lost a full True Strike worth of damage.

Reply:
Fair enough.